### PR TITLE
[openvr] Added Linux support and updated to v1.10.30

### PR DESCRIPTION
**Pull request by makinori**
[ValveSoftware/openvr](https://github.com/ValveSoftware/openvr) contains `linux64` and `linux32` but wasn't being used in the Vcpkg port file so I added them.

`osx32` (no 64 bit and latest version Catalina is 64 bit only) exists too but hasn't been updated in 4 years. I think VR support on Mac is basically non existent but I know Valve is focusing on Linux a lot. 

I updated the version as well.

### CODE REVIEW

File: ports/openvr/portfile.cmake [hunk 1]
```diff
@@ -3,43 +3,39 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/openvr/portfile.cmake:3]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/openvr/portfile.cmake [hunk 2]
```diff
@@ -3,43 +3,39 @@ include(vcpkg_common_functions)
...
     HEAD_REF master
 )
 
 set(VCPKG_LIBRARY_LINKAGE dynamic)
 
-if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-    set(ARCH_PATH "win64")
-elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-    set(ARCH_PATH "win32")
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "")
```
Comment (NancyLi1013) [inline ports/openvr/portfile.cmake:13]: This might be updated as `if(VCPKG_TARGET_IS_WINDOWS)`.

File: ports/openvr/portfile.cmake [hunk 3]
```diff
@@ -3,43 +3,39 @@ include(vcpkg_common_functions)
...
+    if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+        set(ARCH_PATH "win64")
+    elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+        set(ARCH_PATH "win32")
+    else()
+        message(FATAL_ERROR "Package only supports x64 and x86 Windows.")
+    endif()
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (NancyLi1013) [inline ports/openvr/portfile.cmake:21]: This might be updated as `elseif(VCPKG_TARGET_IS_LINUX)`.

File: ports/openvr/portfile.cmake [hunk 4]
```diff
@@ -3,43 +3,39 @@ include(vcpkg_common_functions)
...
-    ${SOURCE_PATH}/bin/${ARCH_PATH}/openvr_api.pdb
-    DESTINATION ${CURRENT_PACKAGES_DIR}/bin
-)
-file(COPY
-    ${SOURCE_PATH}/bin/${ARCH_PATH}/openvr_api.dll
-    ${SOURCE_PATH}/bin/${ARCH_PATH}/openvr_api.pdb
-    DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin
-)
 file(COPY ${SOURCE_PATH}/headers DESTINATION ${CURRENT_PACKAGES_DIR})
 file(RENAME ${CURRENT_PACKAGES_DIR}/headers ${CURRENT_PACKAGES_DIR}/include)
```
Comment (NancyLi1013) [inline ports/openvr/portfile.cmake:41]: Could you please update this
```
file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/openvr)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/openvr/LICENSE ${CURRENT_PACKAGES_DIR}/share/openvr/copyright)
```
as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

File: ports/openvr/CONTROL [hunk 5]
```diff
@@ -1,4 +1,4 @@
 Source: openvr
-Version: 1.9.16
+Version: 1.10.30
 Homepage: https://github.com/ValveSoftware/openvr
 Description: an API and runtime that allows access to VR hardware from multiple vendors without requiring that applications have specific knowledge of the hardware they are targeting.
```
Comment (NancyLi1013) [inline ports/openvr/CONTROL:4]: Could you please help `Supports: !(arm|uwp|osx)` filed here?
Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#supports).

File: ports/openvr/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/openvr/portfile.cmake:3]: It seems that `openvr` currently doesn't support `osx `,`uwp`and `arm`.
Could you please also add `vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp" "osx")` to the top of portfile.cmake?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @makitsune 
Thanks for this PR.
I noticed this port has passed on Linux platform. Could you please remove `openvr:x64-linux=fail` from [ci.baseline.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

**Comment by makitsune** [comment]
Thank you so much @NancyLi1013 for replying so quickly.
I made the changes you suggested.

I didn't replace the `file(COPY ...` calls for `/lib` and `/bin` with `file(INSTALL ...` because the file permissions get reset. Specifically they're not executable (+x) anymore.

Also, `VCPKG_CMAKE_SYSTEM_NAME STREQUAL ""` for detecting Windows doesn't work anymore. Though it's stated in the documentation here:
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_cmake_system_name
Probably best to add that it's deprecated. I could open a pr?

**Comment by NancyLi1013** [comment]
@makitsune 
thanks for your update and reminder.
Currently, we suggest to use `VCPKG_TARGET_IS_WINDOWS, VCPKG_TARGET_IS_LINUX , VCPKG_TARGET_IS_OSX,  VCPKG_TARGET_IS_UWP ` etc to judge the platforms instead of `VCPKG_CMAKE_SYSTEM_NAME `, 
As for the usage of `VCPKG_CMAKE_SYSTEM_NAME  for detecting Windows, it is the correct way to use like this:

`NOT VCPKG_CMAKE_SYSTEM_NAME`

Of course, we need to update the deprecated functions to document. 
You don't need to open a new PR to update this.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**